src/sys/fuchsia/ready.rs

```diff
@@ -0,0 +1,138 @@
+use event_imp::{Ready, ready_as_usize, ready_from_usize};
+pub use magenta_sys::mx_signals_t;
+use std::ops;
+
+// The following impls are valid because Fuchsia and mio both represent
+// "readable" as `1 << 0` and "writable" as `1 << 2`.
+
+/// Fuchsia specific extensions to `Ready`
+///
+/// Provides additional readiness event kinds that are available on Fuchsia.
+///
+/// Conversion traits are implemented between `Ready` and `FuchsiaReady`.
+///
+/// For high level documentation on polling and readiness, see [`Poll`].
+///
+/// [`Poll`]: struct.Poll.html
+#[derive(Debug, Copy, PartialEq, Eq, Clone, PartialOrd, Ord)]
+pub struct FuchsiaReady(Ready);
+
+impl FuchsiaReady {
+    /// Returns whether or not the `FuchsiaReady` contains all of the specified
+    /// magenta signals.
+    #[inline]
+    pub fn has_all(&self, signals: mx_signals_t) -> bool {
+        (Ready::from(signals) - self.0).is_empty()
+    }
+
+    /// Returns whether or not the `FuchsiaReady` contains any of the specified
+    /// magenta signals.
+    #[inline]
+    pub fn has_any(&self, signals: mx_signals_t) -> bool {
```

Comment:
Could this and `has_all` be removed in favor of bit twiddling? (just thinking for consistency between Unix and Fuchsia)